applications/utils/status_manager.py

```diff
@@ -0,0 +1,72 @@
+from applications.models import (
+    Application,
+    ApplicationEvent,
+    ApplicationEventStatus,
+    ApplicationRound,
+    ApplicationStatus,
+)
+
+
```

Comment:
I notice here a lot of model.status = Something => model.save() 
Should we just call the `set_status` method?

Reply:
Good catch. I also fixed the same thing in tests 👍